lib/spack/spack/operating_systems/windows_os.py

```diff
@@ -65,6 +67,8 @@ class WindowsOs(OperatingSystem):
         compiler_search_paths = comp_search_paths
 
     def __init__(self):
+        if platform.release() != '10':
+            raise SpackError("Spack is not supported on Windows versions older than 10")
```

Comment:
There's also Windows 11, so this won't work on that either I assume

Reply:
Spack is entirely untested on Windows 11, although Windows is pretty good about backwards compatibility. I can make this a softer check and just leave a note in the getting started docs. Thoughts?

Reply:
That sounds good: In that case this check should probably be something like `if Version(platform.release()) < Version('10'):`

Reply:
I didn't realize the version class could be used like that. Thanks for the suggestion!